components/authz-service/server/v2/policy_test.go

```diff
@@ -943,15 +1096,19 @@ func TestUpdatePolicy(t *testing.T) {
 
 			require.NoError(t, err)
 
+			// These individual statement checks are included in the final assertPoliciesMatch,
+			// but useful to check these key elements explicitly.
 			s0 := pol.Statements[0]
 			assert.Equal(t, api_v2.Statement_ALLOW, s0.Effect)
 			assert.Equal(t, []string{"compliance:profiles"}, s0.Resources)
 			assert.Equal(t, []string{"compliance:profiles:upload"}, s0.Actions)
+			assert.Equal(t, []string{constants_v2.AllProjectsExternalID}, s0.Projects)
```

Comment:
Implicit or explicit look the same in the end.

Reply:
If the `implicit` way (as in, no projects in the statement means it applies to all projects) still works like before, why do we need to start enforcing projects on all v2.1 statements?

Reply:
But it has never been legal to have a statement with no projects.

Reply:
Do you mean according to the acceptance requirements?